base_app/src/app/app-tab-overlay/overlay-page.ts

```diff
@@ -152,6 +153,31 @@ export class OverlayPageComponent {
         return false;
     }
 
+
+    /**
+     * Returns false if element is not already in elemList
+     * @param elemList list of elements
+     * @param element potenital new element
+     */
+    checkElemListForDuplicates(elemList: any, element: any): boolean {
+        let duplicatesExist = false;
+        const elementID = element.id;
```

Comment:
I think this is ok, but just want to confirm that the element.id is indeed unique.
We allow nearly duplicate elements in different sections by prepending the chevron-encased section name - just checking to make sure this test doesn't disallow that possibility.